### ISSUE TITLE
bat attempts to execute itself when PAGER="bat"

### ISSUE THREAD

**Issue by iravid** [sharkdp/bat#383]
I like having other applications use `bat` as a pager, so I `export PAGER="bat"` in my `.zshrc`. However, `bat` will also use that env var and try to execute itself (recursively, endlessly).

To replicate:
```shell
export PAGER="bat"
# This hangs
ls | bat

# In another shell:
ps -fe | grep bat | wc -l
# 177 on my system

unset PAGER
# Works correctly
ls | bat
```

**Comment by sharkdp** [comment]
Nice find :smile:. Thank you for reporting this.

You can easily work around this by setting `PAGER="bat"`, but using `BAT_PAGER` to override this. For example, you can set `BAT_PAGER="less"`. Starting with the next version of `bat` (or current `master`), you can also set `--pager="less"` in `bat`s config file.

I think we should try to fix this by treating `PAGER="bat"` as if `PAGER` had not been set (and similarly for `BAT_PAGER="bat"` or `--pager="bat"`).

**Comment by iravid** [comment]
Cool. I’ll be happy to try fixing this later this week if no one beats me to it :-)

**Comment by sharkdp** [comment]
Closed via #393 by @rodorgas and 7cac07d84559dc6eb02b643f7f640d81a7a4c07e
